collections/_posts/2022-11-23-Multi_Channel_Stochastic_Variational_Inference.md

```diff
@@ -0,0 +1,100 @@
+    title:   "Multi-Channel Stochastic Variational Inference for the Joint Analysis of Heterogeneous Biomedical Data in Alzheimer's Disease"
+    venue:   "Machine Learning in Clinical Neuroimaging (MLCN), MICCAI Workshop 2018"
+pdf: "https://arxiv.org/pdf/1808.03662.pdf"
+---
+
+
+
+# Notes
+
+* Unsupervised method.
+* **Gaussian linear case** (for validation purposes)
+* Another paper based on deep learning methods was done in 2019 by the same authors: [Sparse Multi-Channel Variational Autoencoder](http://proceedings.mlr.press/v97/antelmi19a/antelmi19a.pdf).
+* A [Github repository](https://gitlab.inria.fr/epione_ML/mcvae) is available for the 2019 paper.
+
+* Link to the [VAE tutorial](https://creatis-myriad.github.io/tutorials/2022-09-12-tutorial-vae.html) 
+
+
+# Highlights
+
+* The aim of this paper is to take into account several types of inputs (which are referred to as _channels_ in the paper) for a better joint analysis.
+* They proposed a novel multi-channel stochastic generative model.
+* Their proposed method can have applications for generic data fusion techniques.
+
+
+# Introduction
+
+When a patient suffers from Alzheimer’s disease, physicians use lots of sources of informations. However, simple univariate correlation analysis are limited in modeling power. To overcome the limitations of mass-univariate analysis, methods such as PLS, RRR, CCA, and their variants (non-linearity, multi-channels) were used in biomedical research. The main issue is that those methods are not generative.
+The authors proposed a novel multi-channel stochastic generative model for the joint analysis of multi-channel heterogeneous data.
+
+
+# Methods
+
+Let $$ \textbf{x}  = \{\textbf{x}_c\}_{c=1}^{C} $$ be a single observation of a set of C channels, where each $$ \textbf{x}_c \in \mathbb{R}^{d_c} $$ is a $$ d_c $$-dimensional vector.
+
+* $$ p(\textbf{z}) $$ : prior
+* $$ p(\textbf{x}_c \vert \textbf{z}, \pmb{θ}_c) $$ : likelihood distribution
+* $$ q(\textbf{z} \vert \textbf{x}_c, \pmb{\phi}_c) $$ : probability density function
+
+The aim is to minimize:
+
+$$ \underset{q∈Q}{argmin} \: \mathbb{E}_c[D_{KL} (q (\textbf{z} \vert \textbf{x}_c, \pmb{\phi}_c) \parallel p (\textbf{z} \vert \textbf{x}_1, \dots , \textbf{x}_C , \pmb{θ}_1, \dots , \pmb{θ}_C ) )] $$
+
+**Hypothesis**: Every channel is conditionally independent from all the others given z (it will allow to factorize the data likelihood, see supplementary material in the paper for more details). 
+
+After developping the equation from above they get:
+
+$$ \mathscr{L}(\textbf{θ}, \pmb{\phi}, \textbf{x}) = \mathbb{E}_c \left[ \mathbb{E}_{q(\textbf{z} \vert \textbf{x}_c )} \left[  \sum_{i=1}^{C} ln \: p(\textbf{x}_i \vert \textbf{z}, \textbf{θ}_c) \right] − D_{KL} (q (\textbf{z} \vert \textbf{x}_c, \pmb{\phi}_c) \parallel p (\textbf{z}) ) \right] $$
+
+
+A figure used in the [Github repository](https://gitlab.inria.fr/epione_ML/mcvae) to understand the general idea:
```

Comment:
Given the discussion and questions regarding the formulation of the objective during the presentation, and following a discussion with Olivier, I propose to provide:
1. An intermediate formulation of the objective
2. More textual interpretations of the equations
```suggestion
The objective can be formulated as:

$$ \underset{q∈Q}{argmin} \: \mathbb{E}_c[D_{KL} (q (\textbf{z} \vert \textbf{x}_c, \pmb{\phi}_c) \parallel p (\textbf{z} \vert \textbf{x}_1, \dots , \textbf{x}_C , \pmb{θ}) )] $$

By rearranging the objective, the authors obtain a lower-bound:

$$
\mathbb{E}_c \left[ \mathbb{E}_{q(\textbf{z} \vert \textbf{x}_c )} \left[  ln \: p(\textbf{x} \vert \textbf{z}, \textbf{θ}_c) \right] − D_{KL} (q (\textbf{z} \vert \textbf{x}_c, \pmb{\phi}_c) \parallel p (\textbf{z}) ) \right]
$$

A few interesting points to note about this new formulation are:
- The **data matching term naturally emerges from reformulating the posterior constraint**, independently from the data matching term commonly imposed by autoencoders;
- The inner expectation enforces the decoders from each channel to accurately reconstruct their respective channels.

Finally, under the hypothesis that channels are conditionally independent from each other given $$\textbf{z}$$, the authors factorize the data likelihood as $$ p(\textbf{x} \vert \textbf{z}, \textbf{θ}_c) = \sum_{i=1}^{C} p(\textbf{x}_i \vert \textbf{z}, \textbf{θ}_c)$$ to reformulate the lower bound like below:

$$ \mathscr{L}(\textbf{θ}, \pmb{\phi}, \textbf{x}) = \mathbb{E}_c \left[ \mathbb{E}_{q(\textbf{z} \vert \textbf{x}_c )} \left[  \sum_{i=1}^{C} ln \: p(\textbf{x}_i \vert \textbf{z}, \textbf{θ}_c) \right] − D_{KL} (q (\textbf{z} \vert \textbf{x}_c, \pmb{\phi}_c) \parallel p (\textbf{z}) ) \right] $$

This **factorization of the data likelihood enforces that the posterior from each channel should be similar**, since they are used to reconstruct the input from other channels.

A figure from the [Github repository](https://gitlab.inria.fr/epione_ML/mcvae) explains schematically the general idea:
```

Reply:
@OlivierBernard Given our discussion earlier today (and if you have the time!) I would appreciate if you could look at this new explanation of objective formulation and tell me if it matches your own interpretation :wink: